### DISCUSSION TITLE
Add an alternative to Basic Authentication so websockets can access phoenixd from a browser

### DISCUSSION THREAD

**Discussion by remyers** [ACINQ/phoenixd#92] [Ideas]
I am working around the basic authentication requirement for websockets with a python [shim](https://github.com/ACINQ/phoenixd/discussions/39#discussioncomment-10308064) so that I can access the websocket from javascript in a local browser. It would be nice if there was an option to do basic authentication via the url instead of the http header.  For example:

```
ws://127.0.0.1:9740/websocket?auth=<http-password-limited-access>
```

Current browsers do not support passing custom header information to websockets, and they are unlikely to ever allow it based on this SO post:  [https://stackoverflow.com/questions/4361173/http-headers-in-websockets-client-api](https://stackoverflow.com/questions/4361173/http-headers-in-websockets-client-api)

**Comment by tsmith123** [comment]
Yeah this would be useful for local webpages and for testing.

**Comment by remyers** [comment]
Fixed by https://github.com/ACINQ/phoenixd/pull/95
